apps/docs/content/guides/ai/examples/semantic-image-search-amazon-titan.mdx

```diff
@@ -0,0 +1,228 @@
+
+```python
+import sys
+import boto3
+import vecs
+import json
+import base64
+from matplotlib import pyplot as plt
+from matplotlib import image as mpimg
+
+DB_CONNECTION = "postgresql://postgres.[PROJECT-REF]:[YOUR-PASSWORD]@aws-0-[REGION].pooler.supabase.com:5432/postgres"
+```
+
+Next, get the [credentials to your AWS account](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html) and instantiate the `boto3` client:
+
+```python
+bedrock_client = boto3.client(
+    'bedrock-runtime',
+    region_name='us-west-2',
+    # Credentials from your AWS account
+    aws_access_key_id='<replace_your_own_credentials>',
+    aws_secret_access_key='<replace_your_own_credentials>',
+    aws_session_token='<replace_your_own_credentials>',
+)
+```
+
+## Create embeddings for your images
+
+In the root of your project, create a new folder called `images` and add some images. You can use the images from the example project on [GitHub](https://github.com/supabase/supabase/tree/master/examples/ai/aws_bedrock_image_search/images) or you can find license free images on [unsplash](https://unsplash.com).
+
+To send images to the Amazon Bedrock API we need to need to encode them as `base64` strings. Create the following helper methods:
+
+```python
+def readFileAsBase64(file_path):
+    try:
+        with open(file_path, "rb") as image_file:
+            input_image = base64.b64encode(image_file.read()).decode("utf8")
+        return input_image
+    except:
+        print("bad file name")
+        sys.exit(0)
+
+
+def construct_bedrock_image_body(base64_string):
+    # https://docs.aws.amazon.com/bedrock/latest/userguide/model-parameters-titan-embed-mm.html
+    return json.dumps(
+        {
+            "inputImage": base64_string,
+            "embeddingConfig": {"outputEmbeddingLength": 1024},
+        }
+    )
+
+
+def get_embedding_from_titan_multimodal(body):
+    response = bedrock_client.invoke_model(
+        body=body,
+        modelId="amazon.titan-embed-image-v1",
+        accept="application/json",
+        contentType="application/json",
+    )
+
+    response_body = json.loads(response.get("body").read())
+    print(response_body)
+    return response_body["embedding"]
+
+
+def encode_image(file_path):
+    base64_string = readFileAsBase64(file_path)
+    body = construct_bedrock_image_body(base64_string)
+    emb = get_embedding_from_titan_multimodal(body)
+    return emb
+```
+
+Next, create a `seed` method, which will create a new Supabase Vector Collection, generate embeddings for your images, and upsert the embeddings into your database:
+
+```python
+def seed():
+    # create vector store client
+    vx = vecs.create_client(DB_CONNECTION)
+
+    # create a collection of vectors with 512 dimensions
+    try:
+        images = vx.create_collection(name="image_vectors", dimension=1024)
+    except:
+        images = vx.get_collection(name="image_vectors")
```

Comment:
These methods were deprecated last year. The new way to do this is 
```python
images = vx.get_or_create_collection(name="image_vectors", dimension=1024)
```

https://supabase.github.io/vecs/0.4/api/#get-or-create-a-collection